frontend/packages/topology/src/components/side-bar/components/SideBarHeading.tsx

```diff
@@ -16,6 +18,11 @@ const SideBarHeading: React.FC<{ element: GraphElement }> = ({ element }) => {
           <TopologyActions element={element} />
         </div>
       </h1>
+      {element.getData().resource.kind === DeploymentConfigModel.kind && (
```

Comment:
The crash happens here whenever the user selects a Helm Chart. The e2e test that fails imports a Helm Chart (the Nodejs one from the Helm catalog) and automatically selects the created Helm Chart.

The problem is that a Helm chart is not k8s resource. (It's mostly saved in a Secret, but there are other options as well.)

```
TypeError: Cannot read properties of undefined (reading 'kind')

Component trace:
at SideBarHeading (http://localhost:12968/static/dev-console/code-refs/topology~overview-list-page-f3b1fecca527617e2d8b.js:3145:27)
```

In this case, `resource` is undefined. But in other cases `getData()` could also return undefined!

So I recommend adding a new variable at the beginning of the component:

```tsx
  const resourceKind = element.getData()?.resource?.kind;
```

And then using this here:

```suggestion
      {resourceKind === DeploymentConfigModel.kind && (
```

Reply:
@jerolimov Hi there :smile: 
Thank you very much for looking into this issue and for the very detailed explanation. :100: 
I'll change my code according to it